src/iceflow/ingest/atm1b.py

```diff
@@ -368,7 +368,24 @@ def _ilatm1bv2_data(fn: Path, file_date: dt.date) -> pd.DataFrame:
     return df
 
 
-def atm1b_data(filepath: Path) -> pd.DataFrame:
+class atm1bData(commonDataColumns):
+    # Data fields unique to ATM1B data.
+    rel_time: Series[pa.dtypes.Int32]
+    xmt_sigstr: Series[pa.dtypes.Int32]
+    rcv_sigstr: Series[pa.dtypes.Int32]
+    azimuth: Series[pa.dtypes.Int32]
+    pitch: Series[pa.dtypes.Int32]
+    roll: Series[pa.dtypes.Int32]
+    gps_pdop: Series[pa.dtypes.Int32]
+    gps_time: Series[pa.dtypes.Int32]
+    passive_signal: Series[pa.dtypes.Int32]
+    passive_footprint_latitude: Series[pa.dtypes.Int32]
+    passive_footprint_longitude: Series[pa.dtypes.Int32]
+    passive_footprint_synthesized_elevation: Series[pa.dtypes.Int32]
+
+
+@pa.check_types()
+def atm1b_data(filepath: Path) -> DataFrame[atm1bData]:
```

Comment:
This looks like PEP695 style generic. I thought you needed to support older Pythons? Does this work if you set Mypy's Python version to 3.9?

Reply:
I'm using py 3.10 as the minimum version. I haven't tested against 3.9.

Reply:
Right, forgot already :) 3.10 also doesn't support this feature. My understanding is that this is OK at runtime because of `from __future__ import annotations`. Does that sound right to you too?

Reply:
Maybe?  I've definitely used this type of syntax on < 3.10 before, without the `from __future__ import annotations`.

Reply:
you're totally right, I'm just confused :laughing: